src/main/java/uk/gov/companieshouse/api/strikeoffobjections/processor/ObjectionProcessor.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.companieshouse.api.strikeoffobjections.processor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.common.LogConstants;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.InvalidObjectionStatusException;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.ObjectionNotFoundException;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.ObjectionStatus;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IObjectionService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Processes an Objection
+ *
+ * Will only process Objection if status = SUBMITTED
+ * Calls Chips to place stop against company
+ * Sends email
+ */
+@Component
+public class ObjectionProcessor {
+
+    private static final String INVALID_START_STATUS_MSG =
+            "Objection %s has status %s. Cannot process unless status = SUBMITTED";
+    private static final String LOG_OBJECTION_ID_KEY = LogConstants.OBJECTION_ID.getValue();
+
+    private IObjectionService objectionService;
+    private ApiLogger apiLogger;
+
+    @Autowired
+    public ObjectionProcessor(IObjectionService objectionService,
+                              ApiLogger apiLogger) {
+        this.objectionService = objectionService;
+        this.apiLogger = apiLogger;
+    }
+
+    /**
+     * Process the specified Objection
+     * Only processes if status is SUBMITTED
+     *
+     * @param httpRequestId http request id used for logging
+     * @param objectionId   id of the objection to process
+     * @throws ObjectionNotFoundException      if objectionId not found in database
+     * @throws InvalidObjectionStatusException if the Objection is not currently in status SUBMITTED when this is called
+     */
+    public void process(String httpRequestId, String objectionId)
+            throws ObjectionNotFoundException, InvalidObjectionStatusException {
+
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(LOG_OBJECTION_ID_KEY, objectionId);
+        apiLogger.infoContext(httpRequestId, "Starting objection processing", logMap);
+
+        Objection objection = getObjectionForProcessing(httpRequestId, objectionId);
+
+        // TODO update status to processing
+
+        // TODO OBJ-139/OBJ-20 do chips sending
+
+        // TODO update status to chips sent
+
+        // TODO OBJ-113 do email sending
+
+        // TODO update status to processed
+
+    }
+
+    private Objection getObjectionForProcessing(String httpRequestId, String objectionId)
+            throws ObjectionNotFoundException, InvalidObjectionStatusException {
+
+        // get objection
+        Objection objection = objectionService.getObjection(httpRequestId, objectionId);
```

Comment:
Perhaps we should refactor the objectionService.patchObjection method to return the result of its repository.save() call. and pass the result into this method to save an uneeded db call

Reply:
We could do that, or another alternative would be to have the ObjectionService.patchObjection call the processor (and pass the entity in to it) and the controller would just call the service? I think @mwestacott had an opinion previously about this as well.

Reply:
How about the controller only calls the service and the service calls the processor, passing in the objection (which it should already have)? @markpit suggested this and I like the idea of keeping the controller nice and simple - delegating everything to the service.

Reply:
ok, i can do that, should get rid of the extra db 'get' call then like Ian said.